logging/google/cloud/logging/logger.py

```diff
@@ -473,6 +517,8 @@ def commit(self, client=None):
                 info = {'protoPayload': MessageToDict(entry)}
             else:
                 raise ValueError('Unknown entry type: %s' % (entry_type,))
+            if resource is not None:
+                info['resource'] = resource._to_dict()
```

Comment:
Yeah that's the idea, explained my thought process above. GLOBAL_RESOURCE is a reasonable default (and default until now). Explicitly setting None is basically saying 'let this entry use the resource of the batch as a whole'. Pretty much what it means in the API.